internal/app/init/pkg/system/service_runner.go

```diff
@@ -336,9 +368,9 @@ func (svcrunner *ServiceRunner) inStateLocked(event StateEvent) bool {
 		health := svcrunner.healthState.Get()
 
 		// up when:
-		//   a) either skipped
+		//   a) either skipped or already finished
 		//   b) or running and healthy (if supports health checks)
-		return svcrunner.state == events.StateSkipped || svcrunner.state == events.StateRunning && (!supportsHealth || (health.Healthy != nil && *health.Healthy))
+		return svcrunner.state == events.StateSkipped || svcrunner.state == events.StateFinished || svcrunner.state == events.StateRunning && (!supportsHealth || (health.Healthy != nil && *health.Healthy))
```

Comment:
I'm fine with it for the sake of this PR, but curious if there's a better way to visually present this instead of a very long set of conditionals? Seems a bit silly to do another case statement, but it might be cleaner to reason about.

Reply:
I wish we could do `svcrunner.state in [Skipped, ...]`, but it's not Python 

yep, I think I could have refactored it to `switch`